client/test/src/main/java/org/elasticsearch/client/RestClientTestUtil.java

```diff
@@ -81,4 +85,22 @@ static int randomErrorRetryStatusCode(Random random) {
     static List<Integer> getAllStatusCodes() {
         return ALL_STATUS_CODES;
     }
+
+    /**
+     * Create the specified number of {@link Header}s.
+     * <p>
+     * Generated header names will be the {@code baseName} plus its index or, rarely, the {@code arrayName} if it's supplied.
```

Comment:
Nit: `arrayName` is not used here any more, not sure if we need it, if not remove from comment.